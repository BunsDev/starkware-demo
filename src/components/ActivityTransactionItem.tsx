import React from 'react';
import { Grid, styled, Typography } from '@material-ui/core';
import { Transaction, TransactionType } from 'context/user';
import mintIcon from 'assets/icons/mint-icon.svg';
import swapIcon from 'assets/icons/swap-icon.svg';

const ActivityListItem = styled('div')(({ theme }) => ({
	height: '81px',
	width: '100%',
	display: 'flex',
	justifyContent: 'space-between',
	boxSizing: 'border-box',
	borderTop: '1px solid rgb(193,193,255,0.2)!important',
	'& div': {
		fontFamily: 'IBM Plex Sans',
		fontStyle: 'normal',
	},
}));
const ActivityDescription = styled('div')(({ theme }) => ({
	height: '80px',
	width: 'fit-content',
	justifyContent: 'flex-start',
	display: 'flex',
	flexDirection: 'row',
}));
const ActivityValue = styled('div')(({ theme }) => ({
	width: 'fit-content',
}));
const ActivityIcon = styled('img')({
	height: '32px',
	width: '32px',
});
const ActivityType = styled('div')({
	height: 'fit-content',
	marginBottom: '13px',

	fontWeight: 'normal',
	fontSize: '18px',
	color: '#FAFAF5',
});
const ActivityDate = styled('div')({
	fontSize: '14px',
	color: '#FAFAF5',
	opacity: 0.6,
});
const Flex = styled('div')({
	display: 'flex',
	justifyContent: 'start',
	alignItems: 'center',
	marginRight: '16px',
});
const ColumnFlex = styled('div')({
	display: 'flex',
	justifyContent: 'center',
	flexDirection: 'column',
	height: '80px',
});

interface ActivityTransactionProps {
	transaction: Transaction;
}
export const ActivityTransactionItem: React.FC<ActivityTransactionProps> = ({
	transaction,
	...props
}) => {
	return (
		<ActivityListItem>
			<ActivityDescription>
				<Flex>
					<ActivityIcon src={transaction.type === TransactionType.MINT ? mintIcon : swapIcon} />
				</Flex>
				<ColumnFlex>
					<ActivityType>{transaction.displayName}</ActivityType>
					<ActivityDate>{transaction.executedOn}</ActivityDate>
				</ColumnFlex>
			</ActivityDescription>
			<ColumnFlex>
				<ActivityValue>{`+${transaction.value} ${transaction.incomingToken.name}`}</ActivityValue>
			</ColumnFlex>
		</ActivityListItem>
	);
};
