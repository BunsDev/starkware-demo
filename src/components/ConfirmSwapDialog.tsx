import React from 'react';
import {
	Button,
	Dialog,
	DialogContent,
	DialogProps,
	DialogTitle,
	Grid,
	makeStyles,
	styled,
	Theme,
	Typography,
} from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';

import { ConversionRate, Token } from 'models/token';
import { ReactComponent as CloseIcon } from '../assets/icons/close.svg';
import { ReactComponent as ArrowDownIcon } from '../assets/icons/arrow-down.svg';
import { ReactComponent as SwapIcon } from '../assets/icons/swap.svg';
import { TokenIcon } from './common/TokenIcon';
import { SwapReceipt } from '../models/swap';

const StyledCloseButton = styled(IconButton)(({ theme }) => ({
	position: 'absolute',
	right: theme.spacing(3),
	top: theme.spacing(4),
	color: theme.palette.grey[500],
	padding: theme.spacing(1),
}));

const StyledDialogTitle = styled(Typography)({
	fontWeight: 600,
	fontSize: 20,
});

const StyledSummaryText = styled(Typography)({
	fontSize: 16,
});

const StyledPrice = styled(StyledSummaryText)({
	opacity: 0.6,
});

const StyledArrowIcon = styled(ArrowDownIcon)({
	paddingLeft: 9,
});

const StyledEndAlignText = styled(Typography)({
	textAlign: 'end',
});

const StyledRow = styled(Grid)({
	marginBottom: 22,
});

const useStyles = makeStyles((theme: Theme) => ({
	root: {
		width: '100%',
		padding: 0,
	},
	dialog: {
		position: 'absolute',
		top: 100,
		borderRadius: theme.spacing(1),
	},
	scrollPaper: {
		alignItems: 'baseline',
	},
	gutters: {
		paddingRight: 33,
		paddingLeft: 33,
	},

	dialogTitle: {
		padding: '30px 33px 35px 33px',
	},
	dialogContent: {
		padding: '0px 33px 33px 33px',
	},
	amount: {
		marginLeft: 13,
		textAlign: 'center',
	},
	rateSummary: {
		justifyContent: 'flex-end',
		[theme.breakpoints.only('xs')]: {
			justifyContent: 'start',
		},
	},
}));

type SwapCandidate = Pick<Token, 'symbol' | 'icon'> & {
	amount: string;
};

interface Props extends DialogProps {
	from?: SwapCandidate;
	to?: SwapCandidate;
	conversionRate: ConversionRate;
	onClose: () => void;
	onSwap: (receipt: SwapReceipt) => void;
}

export const ConfirmSwapDialog = ({ open, from, to, onClose, onSwap }: Props) => {
	const classes = useStyles();

	return (
		<Dialog
			classes={{
				paper: classes.dialog,
				scrollPaper: classes.scrollPaper,
			}}
			fullWidth
			maxWidth="xs"
			open={open}
		>
			<DialogTitle disableTypography className={classes.dialogTitle}>
				<StyledDialogTitle variant="h6">Confirm Swap</StyledDialogTitle>
				<StyledCloseButton aria-label="close" onClick={onClose}>
					<CloseIcon />
				</StyledCloseButton>
			</DialogTitle>
			<DialogContent className={classes.dialogContent}>
				{from && to && (
					<Grid container>
						<StyledRow item container>
							<Grid item container alignItems="center" xs>
								<Grid item>
									<TokenIcon icon={from.icon} />
								</Grid>
								<Grid item className={classes.amount}>
									<Typography variant="body1" color="textPrimary">
										{from.amount}
									</Typography>
								</Grid>
							</Grid>
							<Grid item xs>
								<StyledEndAlignText variant="body1" color="textPrimary">
									{from.symbol}
								</StyledEndAlignText>
							</Grid>
						</StyledRow>
						<StyledRow item container>
							<StyledArrowIcon />
						</StyledRow>
						<StyledRow item container>
							<Grid item container alignItems="center" xs>
								<Grid item>
									<TokenIcon icon={to.icon} />
								</Grid>
								<Grid item className={classes.amount}>
									<Typography variant="body1" color="textPrimary">
										{to.amount}
									</Typography>
								</Grid>
							</Grid>
							<Grid item xs>
								<StyledEndAlignText variant="body1" color="textPrimary">
									{to.symbol}
								</StyledEndAlignText>
							</Grid>
						</StyledRow>
						<StyledRow item container>
							<Grid item container alignItems="center" xs={12} sm>
								<StyledPrice variant="body1" color="textPrimary">
									Price
								</StyledPrice>
							</Grid>
							<Grid
								item
								container
								xs={12}
								sm
								spacing={1}
								className={classes.rateSummary}
								alignItems="center"
							>
								<Grid item>
									<StyledSummaryText variant="body1" color="textPrimary">
										{`${to.amount} ${to.symbol} / ${from.symbol}`}
									</StyledSummaryText>
								</Grid>
								<Grid item>
									<SwapIcon width={13} height={11} />
								</Grid>
							</Grid>
						</StyledRow>
						<Grid item xs>
							<Button
								variant="contained"
								color="secondary"
								fullWidth
								disableElevation
								onClick={() => {
									if (!to) return;
									const { amount, ...token } = to;
									onSwap({ id: '123', token, amount });
								}}
							>
								Confirm Swap
							</Button>
						</Grid>
					</Grid>
				)}
			</DialogContent>
		</Dialog>
	);
};
