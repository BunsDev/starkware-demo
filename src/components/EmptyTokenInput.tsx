import { Button, createStyles, Grid, makeStyles, styled } from '@material-ui/core';
import React from 'react';
import { DarkBox } from './common/DarkBox';
import { TokenIcon } from './common/TokenIcon';
import { NumericInput } from './NumericInput';
import { ReactComponent as TokenSVG } from 'assets/tokens/token.svg';

const InputsContainer = styled(Grid)({
	padding: '0 0 0 12px',
});

const useButtonStyles = makeStyles(() =>
	createStyles({
		root: {
			borderRadius: 50,
		},
		label: {
			fontSize: 12,
      maxWidth: 95,
      height: 16,
		},
	}),
);

export const EmptyTokenInput: React.FC = () => {
  const buttonStyles = useButtonStyles()

	return (
		<DarkBox>
			<Grid container alignItems="center">
				<Grid item>
					<TokenIcon Icon={TokenSVG} size="large" />
				</Grid>
				<InputsContainer item xs>
					<Grid container justify="space-between" alignItems="center">
						<Grid item xs>
							<Button color="secondary" variant="outlined" classes={buttonStyles}>
								Select a Token
							</Button>
						</Grid>
						<Grid item xs>
							<NumericInput onChange={() => false} />
						</Grid>
					</Grid>
				</InputsContainer>
			</Grid>
		</DarkBox>
	);
};
