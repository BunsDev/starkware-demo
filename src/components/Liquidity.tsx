import React from 'react';

export const Liquidity = (): JSX.Element => {
	return <h1>Liquidity</h1>;
};
