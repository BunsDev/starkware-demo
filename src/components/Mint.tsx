import React, { useContext, useState } from 'react';
import { Button, Grid, styled, Typography } from '@material-ui/core';

import { NotificationsContext } from 'context/notifications';
import { Token } from '../models/token';
import { DarkBox } from './common/DarkBox';
import { SelectedToken, TokenSelector } from './TokenSelector';
import { NumericInput } from './NumericInput';
import { useMintError } from '../hooks/amounts';
import { BouncingDots } from './common/BouncingDots';
import { useFilteredTokens } from '../hooks/tokens';
import { useMint } from 'services/API/mutations/useMint';
import { useTxStatus } from 'services/API/queries/useTxStatus';

const StyledInputContainer = styled(Grid)(({ theme }) => ({
	[theme.breakpoints.up('sm')]: {
		paddingLeft: 12,
	},
}));

const StyledContainer = styled(Grid)({
	'& > *': {
		marginBottom: '25.5px',
	},
});

const StyledAddButtonContainer = styled(Grid)({
	'& > *': {
		marginBottom: '10px',
	},
	textAlign: 'center',
});

const StyledAddTokenButton = styled(Button)(({ theme }) => ({
	fontSize: theme.spacing(2),
}));

const StyledLoadingContainer = styled(Grid)({
	height: 236,
	textAlign: 'center',
});

const StyledBouncingDots = styled(BouncingDots)({
	marginBottom: 40,
});

export const Mint = (): JSX.Element => {
	const { dispatch } = useContext(NotificationsContext);
	const { mutate, data: mintData, isLoading: mintLoading } = useMint();
	const { data, error: txError, isLoading, isFetching, isStopped } = useTxStatus(mintData?.tx_id);
	const [mintToken1, setMintToken1] = useState<Token>();
	const [mintToken2, setMintToken2] = useState<Token>();
	const [mintAmount1, setMintAmount1] = useState<string>('1');
	const [mintAmount2, setMintAmount2] = useState<string>('1000');

	const options = useFilteredTokens(mintToken1);
	const mint1Error = useMintError(mintToken1, mintAmount1);
	const mint2Error = useMintError(mintToken2, mintAmount2);
	const error = mint1Error || (mintToken2 && mint2Error);

	const handleAddToken = () => {
		if (!options) return;
		setMintToken2(options[0]);
	};

	const handleMint = () => {
		if (!mintToken1) return;
		mutate({
			accountId: '123',
			token1Amount: '10',
			token2Amount: '256',
		});
		// setLoading(true);
		// setTimeout(() => {
		// 	dispatch({
		// 		type: ActionTypes.OPEN_SUCCESS,
		// 		payload: {
		// 			title: `Success!`,
		// 			icon: mintToken1.icon,
		// 			text: `Received ${mintAmount1} ${mintToken1.symbol}`,
		// 			link: '0xb7d91c4........fa84fc5e6f',
		// 			buttonText: 'Go Back',
		// 		},
		// 	});
		// 	setLoading(false);
		// }, 3000);
	};

	const MintToken1 = () => {
		if (mintToken1 && mintToken2) {
			return <SelectedToken token={mintToken1} />;
		}

		return (
			<TokenSelector
				value={mintToken1}
				options={options}
				onChange={(token) => setMintToken1(token)}
			/>
		);
	};

	if (mintLoading) {
		return (
			<StyledLoadingContainer container alignItems="center" justify="center">
				<Grid item>
					<StyledBouncingDots />
					<Typography color="textPrimary">Loading, Please wait</Typography>
				</Grid>
			</StyledLoadingContainer>
		);
	}

	return (
		<Grid container>
			<StyledContainer item xs={12}>
				<DarkBox>
					<Grid container alignItems="center">
						<Grid item xs aria-label="token to swap">
							<MintToken1 />
						</Grid>
						{mintToken1 && (
							<StyledInputContainer item xs={4} sm={6} alignItems="center">
								<Grid item xs>
									<NumericInput
										inputProps={{
											'aria-label': 'amount of token to swap',
										}}
										value={mintAmount1}
										handleChange={(change) => setMintAmount1(change)}
									/>
								</Grid>
							</StyledInputContainer>
						)}
					</Grid>
				</DarkBox>
			</StyledContainer>
			{mintToken2 && (
				<StyledContainer item xs={12}>
					<DarkBox>
						<Grid container alignItems="center">
							<Grid item xs aria-label="token to swap">
								<SelectedToken token={mintToken2} />
							</Grid>
							<StyledInputContainer item xs={4} sm={6} alignItems="center">
								<Grid item>
									<NumericInput
										inputProps={{
											'aria-label': 'amount of token to swap',
										}}
										value={mintAmount2}
										handleChange={(change) => setMintAmount2(change)}
									/>
								</Grid>
							</StyledInputContainer>
						</Grid>
					</DarkBox>
				</StyledContainer>
			)}
			{mintToken1 && !mintToken2 && (
				<StyledAddButtonContainer item xs={12}>
					<StyledAddTokenButton color="secondary" onClick={handleAddToken}>
						+ Add Token
					</StyledAddTokenButton>
				</StyledAddButtonContainer>
			)}
			<Grid item xs={12}>
				<Button
					variant="contained"
					color="secondary"
					fullWidth
					disableElevation
					disabled={(!mintAmount1 && !mintAmount2) || !!error}
					onClick={handleMint}
				>
					{error ? error : 'Mint'}
				</Button>
			</Grid>
		</Grid>
	);
};
