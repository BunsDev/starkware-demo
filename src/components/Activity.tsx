import React, { useContext, useMemo } from 'react';
import { Grid, styled } from '@material-ui/core';
import { UserContext, TransactionType } from 'context/user';
import { ActivityTransactionItem } from 'components/ActivityTransactionItem';

const ActivityListWrapper = styled('div')({
	maxHeight: '431px',
	height: 'fit-content',
	width: '100%',
	overflowY: 'auto',
});

export const Activity = (): JSX.Element => {
	const { state: userState } = useContext(UserContext);
	const activity = userState.activity;

	const formattedActivity = useMemo(() => {
		return activity
			.map((transaction) => {
				if (transaction.type === TransactionType.MINT) {
					if (transaction.args.mint2) {
						return [
							{
								type: transaction.type,
								timestamp: transaction.timestamp,
								amount: transaction.args.mint1.amount,
								symbol: transaction.args.mint1.token.symbol,
							},
							{
								type: transaction.type,
								timestamp: transaction.timestamp,
								amount: transaction.args.mint2.amount,
								symbol: transaction.args.mint2.token.symbol,
							},
						];
					}

					return {
						type: transaction.type,
						timestamp: transaction.timestamp,
						amount: transaction.args.mint1.amount,
						symbol: transaction.args.mint1.token.symbol,
					};
				}

				return {
					type: transaction.type,
					timestamp: transaction.timestamp,
					amount: transaction.args.to.amount,
					symbol: transaction.args.to.token.symbol,
				};
			})
			.flat();
	}, [activity]);

	return (
		<Grid container>
			<Grid item xs={12}>
				<ActivityListWrapper>
					{formattedActivity.map((transaction, i) => (
						<ActivityTransactionItem key={`transaction-${i}`} {...transaction} />
					))}
				</ActivityListWrapper>
			</Grid>
		</Grid>
	);
};
