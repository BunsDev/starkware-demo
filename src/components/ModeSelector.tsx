import React from 'react';
import { Card, Tabs, Tab, createStyles, Theme, withStyles, CardContent, styled, Grid } from '@material-ui/core';
import { Liquidity } from './Liquidity';
import { Swap } from './Swap';
import { Mint } from './Mint';
import CogIcon from '../assets/icons/cog.svg';

type Mode = 'mint' | 'swap' | 'liquidity';

interface StyledTabProps {
	label: string;
	value: Mode;
}

interface StyledTabsProps {
	value: Mode;
	onChange: (event: React.ChangeEvent<unknown>, mode: Mode) => void;
}

const StyledTabs = withStyles((theme) => ({
	indicator: {
		display: 'flex',
		backgroundColor: 'transparent',
		'& > span': {
			width: '50%',
			backgroundColor: theme.palette.secondary.main,
		},
	},
}))((props: StyledTabsProps) => <Tabs {...props} TabIndicatorProps={{ children: <span /> }} />);

const StyledTab = withStyles((theme: Theme) =>
	createStyles({
		root: {
			height: 20,
			fontSize: 20,
			textTransform: 'none',
			color: 'white',
			marginRight: theme.spacing(1),
			padding: 0,
			'&:focus': {
				opacity: 1,
			},
			[theme.breakpoints.up('lg')]: {
				minWidth: 80,
			},
			[theme.breakpoints.down('lg')]: {
				minWidth: 60,
			},
		},
		wrapper: {
			flexDirection: 'row',
			justifyContent: 'start',
		},
	}),
)((props: StyledTabProps) => <Tab disableRipple {...props} />);

const StyledCard = styled(Card)({
	borderRadius: 8,
	border: '1.5px solid rgba(83, 83, 135, 0.3)',
	padding: '12px 31px 33px',
	boxShadow: '0px 8px 15px 3px rgba(7, 7, 7, 0.13)',
});

const StyledCardContent = styled(CardContent)(({ theme }) => ({
	padding: theme.spacing(3, 0),
}));

export const ModeSelector = () => {
	const [mode, setMode] = React.useState<Mode>('mint');

	const handleModeChange = (_: React.ChangeEvent<unknown>, mode: Mode) => {
		setMode(mode);
	};

	const Content = () => {
		switch (mode) {
			case 'liquidity':
				return <Liquidity />;
			case 'swap':
				return <Swap />;
			case 'mint':
			default:
				return <Mint />;
		}
	};

	return (
		<StyledCard>
			<StyledTabs value={mode} onChange={handleModeChange}>
				<StyledTab label="Mint" value={'mint'} />
				<StyledTab label="Swap" value={'swap'} />
				<StyledTab label="Liquidity" value={'liquidity'} />
				{mode === 'swap' && (
					<Grid container justify="flex-end" xs={12} sm>
						<img src={CogIcon} alt="config-icon" />
					</Grid>
				)}
			</StyledTabs>
			<StyledCardContent>
				<Content />
			</StyledCardContent>
		</StyledCard>
	);
};
