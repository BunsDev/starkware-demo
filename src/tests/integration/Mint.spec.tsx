import React from 'react';
import '@testing-library/jest-dom';

import { customRender, screen, fireEvent, within } from '../utils';
import { Mint } from '../../components/Mint';

describe('Mint', () => {
	const setupMintToken = async () => {
		const { container } = customRender(<Mint />);
		fireEvent.click(screen.getAllByRole('button', { name: /Select a Token/i })[0]);
		const dialog = screen.getByRole('presentation');
		fireEvent.click(within(dialog).getByText(/TK1/i));
		return container;
	};

	it('displays empty token selector initially', () => {
		const { container } = customRender(<Mint />);
		expect(container).toMatchSnapshot();
	});

	it('can select mint token', async () => {
		const container = await setupMintToken();
		expect(container).toMatchSnapshot();
	});

	it('can change mint amount', async () => {
		const container = await setupMintToken();
		fireEvent.change(screen.getByDisplayValue('1000'), {
			target: { value: '500' },
		});
		expect(container).toMatchSnapshot();
	});
});
