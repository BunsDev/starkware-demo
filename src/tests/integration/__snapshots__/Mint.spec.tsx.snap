// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mint can change mint amount 1`] = `
<div>
  <div
    class="MuiGrid-root-1 MuiGrid-container-2"
  >
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiBox-root-105 MuiBox-root-106 Styled(MuiBox)-root-104"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
        >
          <div
            aria-label="token to swap"
            class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
            >
              <div
                class="MuiGrid-root-1 MuiGrid-item-3"
              >
                <div
                  class="MuiBox-root-105 MuiBox-root-325 Styled(MuiBox)-root-294"
                >
                  <div
                    class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-align-items-xs-center-10"
                  >
                    <div
                      class="MuiGrid-root-1 MuiGrid-item-3"
                    >
                      <div
                        class="MuiAvatar-root-276 MuiAvatar-circle-278 WithStyles(ForwardRef(Avatar))-root-274 WithStyles(ForwardRef(Avatar))-root-326 MuiAvatar-colorDefault-277"
                        dimension="48"
                      >
                        <img
                          alt="token icon"
                          src="token.svg"
                        />
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
                    >
                      <div
                        class="MuiBox-root-105 MuiBox-root-328 Styled(MuiBox)-root-297 Styled(MuiBox)-root-327"
                      >
                        <p
                          class="MuiTypography-root-214 WithStyles(ForwardRef(Typography))-root-300 MuiTypography-body1-216 MuiTypography-colorTextPrimary-239"
                        >
                          TK1
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root-1 MuiGrid-item-3"
                    >
                      <button
                        aria-label="change token"
                        class="MuiButtonBase-root-138 MuiIconButton-root-246"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label-254"
                        >
                          <svg>
                            dropdown-arrow.svg
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root-172"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-301 WithStyles(ForwardRef(Grid))-root-302 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-align-items-xs-center-10 MuiGrid-grid-xs-4-39 MuiGrid-grid-sm-6-55"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
            >
              <div
                autocorrect="off"
                class="MuiInputBase-root-304  makeStyles-input-303 MuiInputBase-fullWidth-314"
                inputmode="decimal"
                spellcheck="false"
              >
                <input
                  aria-label="amount of token to swap"
                  autocomplete="off"
                  class="MuiInputBase-input-315"
                  pattern="^[0-9]*[.,]?[0-9]*$"
                  placeholder="0"
                  type="text"
                  value="500"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-322 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <button
        class="MuiButtonBase-root-138 MuiButton-root-109 MuiButton-text-111 WithStyles(ForwardRef(Button))-root-323 WithStyles(ForwardRef(Button))-root-324 MuiButton-textSecondary-113"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-110"
        >
          + Add Token
        </span>
        <span
          class="MuiTouchRipple-root-172"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <button
        class="MuiButtonBase-root-138 MuiButton-root-109 MuiButton-contained-117 MuiBox-root-105 MuiBox-root-161 MuiButton-containedSecondary-119 MuiButton-disableElevation-120 MuiButton-fullWidth-132"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-110"
        >
          Mint
        </span>
        <span
          class="MuiTouchRipple-root-172"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Mint can select mint token 1`] = `
<div>
  <div
    class="MuiGrid-root-1 MuiGrid-container-2"
  >
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiBox-root-105 MuiBox-root-106 Styled(MuiBox)-root-104"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
        >
          <div
            aria-label="token to swap"
            class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
            >
              <div
                class="MuiGrid-root-1 MuiGrid-item-3"
              >
                <div
                  class="MuiBox-root-105 MuiBox-root-295 Styled(MuiBox)-root-294"
                >
                  <div
                    class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-align-items-xs-center-10"
                  >
                    <div
                      class="MuiGrid-root-1 MuiGrid-item-3"
                    >
                      <div
                        class="MuiAvatar-root-276 MuiAvatar-circle-278 WithStyles(ForwardRef(Avatar))-root-274 WithStyles(ForwardRef(Avatar))-root-296 MuiAvatar-colorDefault-277"
                        dimension="48"
                      >
                        <img
                          alt="token icon"
                          src="token.svg"
                        />
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
                    >
                      <div
                        class="MuiBox-root-105 MuiBox-root-299 Styled(MuiBox)-root-297 Styled(MuiBox)-root-298"
                      >
                        <p
                          class="MuiTypography-root-214 WithStyles(ForwardRef(Typography))-root-300 MuiTypography-body1-216 MuiTypography-colorTextPrimary-239"
                        >
                          TK1
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root-1 MuiGrid-item-3"
                    >
                      <button
                        aria-label="change token"
                        class="MuiButtonBase-root-138 MuiIconButton-root-246"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label-254"
                        >
                          <svg>
                            dropdown-arrow.svg
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root-172"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-301 WithStyles(ForwardRef(Grid))-root-302 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-align-items-xs-center-10 MuiGrid-grid-xs-4-39 MuiGrid-grid-sm-6-55"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
            >
              <div
                autocorrect="off"
                class="MuiInputBase-root-304  makeStyles-input-303 MuiInputBase-fullWidth-314"
                inputmode="decimal"
                spellcheck="false"
              >
                <input
                  aria-label="amount of token to swap"
                  autocomplete="off"
                  class="MuiInputBase-input-315"
                  pattern="^[0-9]*[.,]?[0-9]*$"
                  placeholder="0"
                  type="text"
                  value="1000"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-322 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <button
        class="MuiButtonBase-root-138 MuiButton-root-109 MuiButton-text-111 WithStyles(ForwardRef(Button))-root-323 WithStyles(ForwardRef(Button))-root-324 MuiButton-textSecondary-113"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-110"
        >
          + Add Token
        </span>
        <span
          class="MuiTouchRipple-root-172"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <button
        class="MuiButtonBase-root-138 MuiButton-root-109 MuiButton-contained-117 MuiBox-root-105 MuiBox-root-161 MuiButton-containedSecondary-119 MuiButton-disableElevation-120 MuiButton-fullWidth-132"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-110"
        >
          Mint
        </span>
        <span
          class="MuiTouchRipple-root-172"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Mint displays empty token selector initially 1`] = `
<div>
  <div
    class="MuiGrid-root-1 MuiGrid-container-2"
  >
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiBox-root-105 MuiBox-root-106 Styled(MuiBox)-root-104"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
        >
          <div
            aria-label="token to swap"
            class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
            >
              <div
                class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-align-items-xs-center-10"
              >
                <div
                  class="MuiGrid-root-1 MuiGrid-item-3"
                >
                  <svg>
                    placeholder.svg
                  </svg>
                </div>
                <div
                  class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
                >
                  <button
                    class="MuiButtonBase-root-138 MuiButton-root-109 makeStyles-root-107 MuiButton-outlined-114 MuiButton-outlinedSecondary-116"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-110 makeStyles-label-108"
                    >
                      Select a Token
                    </span>
                    <span
                      class="MuiTouchRipple-root-162"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <button
        class="MuiButtonBase-root-138 MuiButton-root-109 MuiButton-contained-117 MuiBox-root-105 MuiBox-root-161 MuiButton-containedSecondary-119 MuiButton-disableElevation-120 MuiButton-disabled-122 MuiButton-fullWidth-132 MuiButtonBase-disabled-139"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label-110"
        >
          Select a token
        </span>
      </button>
    </div>
  </div>
</div>
`;
